scipy/signal/_bsplines.py

```diff
@@ -369,6 +384,118 @@ def qspline1d(signal, lamb=0.0):
         return _quadratic_coeff(signal)
 
 
+def collapse_2d(x, axis):
+    x = moveaxis(x, axis, -1)
+    x_shape = x.shape
+    x = x.reshape(-1, x.shape[-1])
+    if not x.flags.c_contiguous:
+        x = x.copy()
+    return x, x_shape
+
+
+def symiirorder_nd(func, input, *args, axis=-1, **kwargs):
+    axis = normalize_axis_index(axis, input.ndim)
+    input_shape = input.shape
+    input_ndim = input.ndim
+    if input.ndim > 1:
+        input, input_shape = collapse_2d(input, axis)
+
+    out = func(input, *args, **kwargs)
+
+    if input_ndim > 1:
+        out = out.reshape(input_shape)
+        out = moveaxis(out, -1, axis)
+        if not out.flags.c_contiguous:
+            out = out.copy()
+    return out
+
+
+def qspline2d(signal, lamb=0.0, precision=-1.0):
+    """
+    Coefficients for 2-D quadratic (2nd order) B-spline.
+
+    Return the second-order B-spline coefficients over a regularly spaced
+    input grid for the two-dimensional input image.
+
+    Parameters
+    ----------
+    input : ndarray
+        The input signal.
+    lamb : float
+        Specifies the amount of smoothing in the transfer function.
+    precision : float
+        Specifies the precision for computing the infinite sum needed to apply
+        mirror-symmetric boundary conditions.
+
+    Returns
+    -------
+    output : ndarray
+        The filtered signal.
+    """
+    if precision == -1.0:
+        if signal.dtype.char in {'f', 'F'}:
+            precision = 1e-3
+        else:
+            precision = 1e-6
```

Comment:
This was https://github.com/scipy/scipy/pull/18926/files#diff-1bcc32e1c74a323eb44f3afc3cd2a9d3b4b64c32654b9e775c7969daa33e5973L76 and simiar for doubles, correct? Then for backwards compat could you please keep `if precision < -1 or precision > 1`.

While at it, might change `dtype.char` to `if signal.dtype in [np.foat32, np.complex64]`.